src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -335,11 +335,25 @@ class DataResourceDataProvider implements IGridDataProvider {
 			if (endIndex > maxRow) {
 				endIndex = maxRow;
 			}
-			let result = this.rows.slice(index, endIndex).map(row => {
+			let result: ICellValue[][] = [];
+			if (includeHeaders) {
+				let columnHeaders: ICellValue[] = columns.map(col => {
```

Comment:
If you're using map you should be using the return value for something. Either use forEach instead or concat the result onto the result array (I prefer the second slightly)